src/context/ProductContext.js

```diff
@@ -0,0 +1,39 @@
+import { createContext, useContext, useReducer, useEffect } from "react";
+import axios from "axios";
+import { productReducer } from "../reducers";
+
+const ProductContext = createContext(null);
+
+const initialState = {
+  products: [],
+  loading: false,
+  error: null,
+};
+
+const ProductProvider = ({ children }) => {
+  const [state, dispatch] = useReducer(productReducer, initialState);
```

Comment:
Hey Abhishek, I have a question. So is it really necessary to use useReducer here? As the operations are quite simple so useState will also do the job easily.

Reply:
No, we can do the same with useState also. 
I'm doing this way for two reasons - 
1. I want to build a mental model for the context + reducers. So that I'll be more comfortable with redux.
2. useState is pretty easy & straightforward. But reducers, actions with context need more practice. I'm using useState for local state only.

Reply:
Ok. So it is more related to creating the mental model. Thanks, Got it.!!